apps/builder/app/builder/features/sidebar-left/navigator/css-preview.tsx

```diff
@@ -21,33 +22,62 @@ const preStyle = css(textVariants.mono, {
   cursor: "text",
 });
 
+// - Compiles a CSS string from the style info
+// - Groups by category and separates categories with comments
 const getCssText = (instanceStyle: StyleInfo) => {
   const cssEngine = createCssEngine();
-  const style: Style = {};
+  type StyleValueInfoMap = Map<
+    StyleProperty,
+    StyleValueInfo[keyof StyleValueInfo]
+  >;
+  const sourceStyles: StyleValueInfoMap = new Map();
+  const inheritedStyles: StyleValueInfoMap = new Map();
+  const cascadedStyles: StyleValueInfoMap = new Map();
+  const presetStyles: StyleValueInfoMap = new Map();
+
   let property: StyleProperty;
   for (property in instanceStyle) {
     const value = instanceStyle[property];
+    if (value === undefined) {
+      continue;
+    }
+    if (value.local ?? value.nextSource?.value ?? value.previousSource?.value) {
+      sourceStyles.set(property, value.value);
+      continue;
+    }
+    if (value.preset) {
+      presetStyles.set(property, value.value);
+      continue;
+    }
+    if (value.cascaded?.value) {
+      cascadedStyles.set(property, value.value);
+      continue;
+    }
+    if (value.inherited?.value) {
+      inheritedStyles.set(property, value.value);
+      continue;
+    }
+  }
```

Comment:
What if we get here? Should be some exchaustive check if something will change in the future

Reply:
please explain

Reply:
You have a huge series of if's
In some places you explicitly do nothing like 
```
 if (value === undefined) {
      continue;
    }
```
At the end of series you or should cover all possibilities, and have check that code will not be occured there.
For example we will add additional style etc 
Example https://github.com/webstudio-is/webstudio-builder/blob/61f164d40e6c62bc98a0f9f5cb894984b2d93bd5/packages/css-engine/src/core/to-value.ts#L7

Reply:
or give a comment that other possibilities exists and it's not an error that you are doing nothing

Reply:
I am confused, ofc its its not an error, we are exiting early beause the value was handled, its a typical early return situation though with a continue ... I don't know what can be misunderstood

Reply:
Its not misunderstanding its error.
See example I gave with `assertUnreachable` every time we add new style type my code give ts error.
Your code if we will add additional source will never somehow alert us that new style source exists and must be fixed here